UnitySDK/Assets/ML-Agents/Scripts/Policy/RemotePolicy.cs

```diff
@@ -23,10 +23,9 @@ public class RemotePolicy : IPolicy
             string behaviorName)
         {
             m_BehaviorName = behaviorName;
-            var aca = Object.FindObjectOfType<Academy>();
-            aca.LazyInitialization();
-            m_Communicator = aca.Communicator;
-            aca.Communicator.SubscribeBrain(m_BehaviorName, brainParameters);
+            Academy.Instance.LazyInitialization();
```

Comment:
Is LazyInitialization still needed now that there is a Lazy<Academy> ?

Reply:
It's a no-op the first time, but you need to re-initialize if you de-initialize.

Reply:
How does deintialization work? Shouldn't that destroy the singleton instance (and then another gets re-initialized when it is recreated)?

Reply:
Now the singleton is deinitialized on Academy.Dispose